### DISCUSSION TITLE
force_legacy_domain=1 added automatically in URL

### DISCUSSION THREAD

**Discussion by Anshu-Shar** [kinngh/shopify-node-express-mongodb-app#104] [General]
Hello @kinngh ,
I need your help regarding the app redirection issue in my PHP app.  I have created the app in PHP and using the phpish library How can I redirect the app URL correctly because it is automatically adding the ?force_legacy_domain=1. I checked the response -code=28dc4a2827a22423eaeddfdd13e3ceaf&hmac=148cbcce7b89d1953fdca166557d395039a22484020105a26808304a5496f1ba&host=YWRtaW4uc2hvcGlmeS5jb20vc3RvcmUvZ3dsLWFwcHMtZGVtbw&shop=storename.myshopify.com&timestamp=1694701999.I am receiving the correct response.  I searched last 70-80 hours to resolve the issue. But didn't get any help from the forum and community. I know you are working on the MERN.so you can provide me only the guidance for the same. Please help me out. Thanks in advance.
Please check the video for this issue:-
https://www.awesomescreenshot.com/video/21076324?key=887d3bf561729d33ae9d3832548e6c9e

**Comment by kinngh** [comment]
I actually have 0 idea about the PHP ecosystem tbh with you. Let me read up a little.

**Reply by Anshu-Shar** [reply]
Okay,Thank You for your reply

**Reply by kinngh** [reply]
So I couldn't find anything that'd make sense to me, BUT, have you upgraded all your packages like AppBridge and how you're running OAuth?

**Reply by Anshu-Shar** [reply]
`**Hi,` I am using the phpish library for the app. For authentication, I am using the below oauth.php code-**


> 

- [ ] <?php
- [ ]     session_start();
- [ ]     require __DIR__.'/vendor/autoload.php';
- [ ]     use phpish\shopify;

- [ ]     require __DIR__.'/conf.php';
- [ ]     require __DIR__.'/common_functions.php';
- [ ]     # Guard: http://docs.shopify.com/api/authentication/oauth#verification

- [ ]     shopify\is_valid_request_hmac($_GET, SHOPIFY_APP_SHARED_SECRET) or die('Invalid Request! Request or redirect did not come from Shopify');
- [ ]     # Step 2: http://docs.shopify.com/api/authentication/oauth#asking-for-permission

- [ ] header("Content-Security-Policy: frame-ancestors https://" . $_GET['shop'] . " https://admin.shopify.com");
- [ ] if (!isset($_GET['code'])) {
- [ ]     $permission_url = shopify\authorization_url($_GET['shop'], SHOPIFY_APP_API_KEY, array('read_products', 'write_products','read_orders'), REDIRECT_URL);
- [ ]     die("<script> top.location.href='$permission_url'</script>");
}
- [ ]     # Step 3: http://docs.shopify.com/api/authentication/oauth#confirming-installation
- try {
- [ ]     # shopify\access_token can throw an exception

- [ ]     $oauth_token = shopify\access_token($_GET['shop'], SHOPIFY_APP_API_KEY, SHOPIFY_APP_SHARED_SECRET, $_GET['code']);
- [ ]     $_SESSION['oauth_token'] = $oauth_token;
- [ ]     $_SESSION['shop'] = $_GET['shop'];
- [ ]     $host=$_GET['host'];
- [ ]     $shop_name = $_GET['shop'];
- [ ]     $oauth_token_check = get_oauth_details($shop_name);
- [ ]     if ($oauth_token_check == '') {
- [ ]         $oauth_token = $_SESSION['oauth_token'];
- [ ]         insert_oauth_details($shop_name, $oauth_token);
- [ ]     } else {
- [ ]         update_oauth_details($shop_name, $oauth_token);
- [ ]     }
 
- [ ]     $current_script_name = split("/", $_SERVER['SCRIPT_NAME']);
- [ ]     if (($key = array_search(end($current_script_name), $current_script_name)) !== false) {
- [ ]         unset($current_script_name[$key]);
- [ ]     }
- [ ]      $app_path = implode("/", $current_script_name);
- [ ]      $shop_name=trim(str_replace('.myshopify.com', '', $shop_name));

- [ ]      $decodedValue = base64_decode($host);
- [ ]       $redirection_url = "https://$decodedValue/apps/".SHOPIFY_APP_API_KEY."$app_path/admin.php";

- [ ]     die("<script> top.location.href='$redirection_url'</script>");
- [ ] } catch (shopify\ApiException $e) {
- [ ]     # HTTP status code was >= 400 or response contained the key 'errors'
- [ ]     echo $e;
- [ ]     print_r($e->getRequest());
- [ ]     print_r($e->getResponse());
 } catch (shopify\CurlException $e) {
- [ ]     # cURL error
- [ ]     echo $e;
- [ ]     print_r($e->getRequest());
- [ ]     print_r($e->getResponse());
- [ ] }

**Reply by kinngh** [reply]
Ok I want you to understand I'm not a PHP developer and your code isn't properly formatted either. I'm a JS/Rust dev and PHP means nothing to me^

**Reply by Anshu-Shar** [reply]
Ok, Thanks once again
